benchmarks/strings/bench_strings.sh

```diff
@@ -20,375 +20,263 @@ source ../bench_plot.sh
-		fi
-		bench_command $i buf_ropes_iter$i ./iteration_bench -m ropebuf --iter-mode iterator --loops $2 --strlen $3 --ccts $i
-	done
 
-	prepare_res iter_buf_ropes_index.out iter_buf_ropes_index buf_ropes_index
-	if $verbose; then
-		echo "RopeBuffer by index :"
-	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String base length = $1, Concats (depth of the rope) = $i, Loops = $3"
+	isok=false
+	for i in $benches; do
+		if [ $1 = $i ]; then
+			isok=true;
 		fi
-		bench_command $i buf_ropes_index$i ./iteration_bench -m ropebuf --iter-mode index --loops $2 --strlen $3 --ccts $i
 	done
-
-	plot iter.gnu
-
-	cd ..
-}
-
-function bench_substr()
-{
-	if [ -d string_substr ]; then
-		rm string_substr/*
-	else
-		mkdir string_substr
+	if [ "$isok" = false ]; then
+		usage;
+		exit;
 	fi
-	cd string_substr
 
-	if $verbose; then
-		echo "*** Benching substring performance ***"
-	fi
+	bench=$1
+	shift;
 
-	../../../bin/nitc --global ../substr_bench.nit
+	head=`git rev-parse HEAD`
 
-	prepare_res substr_flat.out substr_flat flatstring
-	if $verbose; then
-		echo "FlatStrings :"
+	if [ "$bench" = "basic" ]; then
+		basic;
+		exit;
 	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String length = $i, loops = $2, Loops = $3"
-		fi
-		bench_command $i flatstring$i ./substr_bench -m flatstr --loops $2 --strlen $3 --ccts $i
-	done
 
-	prepare_res substr_buf.out substr_buf flatbuffer
-	if $verbose; then
-		echo "FlatBuffers :"
+	if [ ! -d out ]; then
+		mkdir out
 	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String length = $i, loops = $2, Loops = $3"
-		fi
-		bench_command $i flatbuffer$i ./substr_bench -m flatbuf --loops $2 --strlen $3 --ccts $i
-	done
+	cd out
 
-	prepare_res substr_ropes.out substr_ropes ropes
-	if $verbose; then
-		echo "Ropes :"
+	if [ -d $bench ]; then
+		rm $bench/*
+	else
+		mkdir $bench
 	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String length = $i, loops = $2, Loops = $3"
-		fi
-		bench_command $i ropes$i ./substr_bench -m ropestr --loops $2 --strlen $3 --ccts $i
-	done
+	cd ..
 
-	prepare_res substr_buf_ropes.out substr_buf_ropes buf_ropes
-	if $verbose; then
-		echo "RopeBuffers :"
-	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String length = $i, loops = $2, Loops = $3"
-		fi
-		bench_command $i buf_ropes$i ./substr_bench -m ropebuf --loops $2 --strlen $3 --ccts $i
+	echo "---------------------------------------------------------"
+	echo "  Trying variant HEAD for benchmark $bench"
+	echo "---------------------------------------------------------"
+	prepare_compiler
+	bench_$bench $@;
+
+	for i in lib_variants/*; do
+		variant=`basename "$i" | cut -f 1 -d '.'`
+		echo "---------------------------------------------------------"
+		echo "  Trying variant $variant for benchmark $bench"
+		echo "---------------------------------------------------------"
+		git am $i
+		prepare_compiler
```

Comment:
Why do you genererate the compiler? You test libs variations, the same compiler should be used, only the lib should vary.